File: 2215. Find the Difference of Two Arrays/main.rs
struct Solution;

use std::collections::HashSet;

impl Solution {
    pub fn find_difference(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<Vec<i32>> {
        let mut set1: HashSet<i32> = HashSet::from_iter(nums1.iter().cloned());
        let mut set2: HashSet<i32> = HashSet::from_iter(nums2.iter().cloned());

        for num in nums1.iter() {
            if set2.contains(num) {
                set1.remove(num);
                set2.remove(num);
            }
        }

        vec![set1.into_iter().collect(), set2.into_iter().collect()]
    }
}

use std::collections::HashSet;

impl Solution {
    pub fn find_difference(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<Vec<i32>> {
        let set1: HashSet<i32> = nums1.into_iter().collect();
        let set2: HashSet<i32> = nums2.into_iter().collect();
        let diff1 = set1
            .iter()
            .filter(|num| !set2.contains(num))
            .cloned()
            .collect();
        let diff2 = set2
            .iter()
            .filter(|num| !set1.contains(num))
            .cloned()
            .collect();

        vec![diff1, diff2]
    }
}

File: 2215. Find the Difference of Two Arrays/main.py
from typing import List


class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        set1 = set(nums1)
        set2 = set(nums2)

        for num in nums1:
            if num in set2:
                set1.discard(num)
                set2.discard(num)

        return [list(set1), list(set2)]


class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        set1 = set(nums1)
        set2 = set(nums2)

        return [list(set1 - set2), list(set2 - set1)]
